reflex/app.py

```diff
@@ -211,10 +209,7 @@ def __init__(self, *args, **kwargs):
         self.setup_admin_dash()
 
         # If a State is not used and no overlay_component is specified, do not render the connection modal
-        if (
-            self.state is DefaultState
-            and self.overlay_component is default_overlay_component
-        ):
+        if not self.state and self.overlay_component is default_overlay_component:
```

Comment:
```suggestion
        if self.state is None and self.overlay_component is default_overlay_component:
```

I'd prefer to see the explicit `None` check here